### PR TITLE
[Merged by Bors] - feat: Set explicit resources on all containers

### PR DESCRIPTION
**Pull request by razvan**
# Description

Part of https://github.com/stackabletech/issues/issues/394

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes
 
```[tasklist]
# Author
- [ ] Changes are OpenShift compatible
- [ ] CRD changes approved
- [ ] Helm chart can be installed and deployed operator works
- [ ] Integration tests passed (for non trivial changes)
```

```[tasklist]
# Reviewer
- [ ] Code contains useful comments
- [ ] (Integration-)Test cases added
- [ ] Documentation added or updated
- [ ] Changelog updated
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
```

Once the review is done, comment `bors r+` (or `bors merge`) to merge. [Further information](https://bors.tech/documentation/getting-started/#reviewing-pull-requests)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,7 @@
 - Fix LDAP tests for Openshift ([#270]).
 - Missing CRD defaults for `status.conditions` field ([#277]).
 - Support Airflow `2.6.1` ([#284]).
+- Add support for resource quota ([#289])
```
Comment (sbernauer) [inline CHANGELOG.md:11]:
```suggestion
-  Set explicit resources on all containers ([#289])
```

File: rust/crd/src/lib.rs [hunk 2]
```diff
@@ -487,11 +487,11 @@ impl AirflowConfig {
         AirflowConfigFragment {
             resources: ResourcesFragment {
                 cpu: CpuLimitsFragment {
-                    min: Some(Quantity("200m".to_owned())),
-                    max: Some(Quantity("4".to_owned())),
+                    min: Some(Quantity("100m".to_owned())),
```
Comment (sbernauer) [inline rust/crd/src/lib.rs:490]: Attention: We might need different defaults for different roles (scheduler, worker etc.)

**Review by sbernauer — COMMENTED** [review]
LGTM, but I would like to re-discuss the actual default resource values via Slack

**Review by sbernauer — APPROVED** [review]
LGTM when tests pass

File: tests/templates/kuttl/smoke/helm-bitnami-redis-values.yaml.j2 [hunk 3]
```diff
@@ -13,6 +13,13 @@ master:
 {% endif %}
   containerSecurityContext:
     enabled: false
+  resources:
+    requests:
+      memory: "128Mi"
+      cpu: "512m"
+    limits:
+      memory: "128Mi"
+      cpu: "1"
```
Comment (sbernauer) [inline tests/templates/kuttl/smoke/helm-bitnami-redis-values.yaml.j2:22]:
```suggestion
      cpu: "100m"
    limits:
      memory: "128Mi"
      cpu: "400m"
```

File: tests/templates/kuttl/smoke/helm-bitnami-redis-values.yaml.j2 [hunk 4]
```diff
@@ -24,6 +31,13 @@ replica:
 {% endif %}
   containerSecurityContext:
     enabled: false
+  resources:
+    requests:
+      memory: "128Mi"
+      cpu: "100m"
+    limits:
+      memory: "128Mi"
+      cpu: "512m"
```
Comment (sbernauer) [inline tests/templates/kuttl/smoke/helm-bitnami-redis-values.yaml.j2:40]:
```suggestion
      cpu: "100m"
    limits:
      memory: "128Mi"
      cpu: "400m"
```

File: tests/templates/kuttl/smoke/50-install-airflow-python.yaml [hunk 5]
```diff
@@ -20,3 +20,10 @@ spec:
           image: docker.stackable.tech/stackable/testing-tools:0.2.0-stackable0.0.0-dev
           stdin: true
           tty: true
+          resources:
+            requests:
+              memory: "128Mi"
+              cpu: "100m"
+            limits:
+              memory: "128Mi"
+              cpu: "512m"
```
Comment (sbernauer) [inline tests/templates/kuttl/smoke/50-install-airflow-python.yaml:29]:
```suggestion
              cpu: "400m"
```

**Review by sbernauer — APPROVED** [review]

**Comment by sbernauer** [comment]
I think the webserver needs more memory. Don't forget to update the docs ;)
image

**Comment by Techassi** [comment]
Increased the webserver memory request/limit to `1024Mi`.

**Comment by Techassi** [comment]
Jenkins tests pass, see custom test run `#90`:

```
--- PASS: kuttl (4089.72s)
    --- PASS: kuttl/harness (0.00s)
        --- PASS: kuttl/harness/logging_airflow-2.2.4-stackable0.0.0-dev_openshift-false (254.02s)
        --- PASS: kuttl/harness/ldap_airflow-latest-2.6.1-stackable0.0.0-dev_ldap-authentication-insecure-tls_openshift-false (281.80s)
        --- PASS: kuttl/harness/smoke_airflow-2.6.1-stackable0.0.0-dev_openshift-false (278.62s)
        --- PASS: kuttl/harness/smoke_airflow-2.4.1-stackable0.0.0-dev_openshift-false (290.70s)
        --- PASS: kuttl/harness/smoke_airflow-2.2.5-stackable0.0.0-dev_openshift-false (324.53s)
        --- PASS: kuttl/harness/smoke_airflow-2.2.4-stackable0.0.0-dev_openshift-false (302.84s)
        --- PASS: kuttl/harness/mount-dags-configmap_airflow-latest-2.6.1-stackable0.0.0-dev_openshift-false (242.95s)
        --- PASS: kuttl/harness/ldap_airflow-latest-2.6.1-stackable0.0.0-dev_ldap-authentication-server-verification-tls_openshift-false (298.18s)
        --- PASS: kuttl/harness/ldap_airflow-latest-2.6.1-stackable0.0.0-dev_ldap-authentication-no-tls_openshift-false (233.20s)
        --- PASS: kuttl/harness/cluster-operation_airflow-latest-2.6.1-stackable0.0.0-dev_openshift-false (281.56s)
        --- PASS: kuttl/harness/resources_airflow-latest-2.6.1-stackable0.0.0-dev_openshift-false (168.83s)
        --- PASS: kuttl/harness/orphaned-resources_airflow-latest-2.6.1-stackable0.0.0-dev_openshift-false (225.28s)
        --- PASS: kuttl/harness/mount-dags-gitsync_airflow-latest-2.6.1-stackable0.0.0-dev_openshift-false (272.55s)
        --- PASS: kuttl/harness/logging_airflow-2.4.1-stackable0.0.0-dev_openshift-false (195.94s)
        --- PASS: kuttl/harness/logging_airflow-2.6.1-stackable0.0.0-dev_openshift-false (230.88s)
        --- PASS: kuttl/harness/logging_airflow-2.2.5-stackable0.0.0-dev_openshift-false (207.33s)
```

**Pull request closed without merging**